compiler/semobjconstr.nim

```diff
@@ -84,6 +84,66 @@ proc caseBranchMatchesExpr(branch, matched: PNode): bool =
 
   return false
 
+type
+  Pairs = seq[(BiggestInt, BiggestInt)]
+
+proc isSubsetOf(lesser, greater: Pairs): bool =
+  result = true
+  for i in 0 ..< lesser.len:
+    block checkNextSubset:
+      for j in 0 ..< greater.len:
+        if lesser[i][0] >= greater[j][0] and lesser[i][1] <= greater[j][1]:
+          break checkNextSubset
+      return false
+
+proc addBranchVals(s: var Pairs, b: PNode) =
+  if b.kind != nkElifBranch:
+    for i in 0 .. b.len-2:
+      if b[i].kind == nkIntLit:
+        s.add (b[i].intVal, b[i].intVal)
+      elif b[i].kind == nkRange:
+        s.add (b[i][0].intVal, b[i][1].intVal)
+
+proc mergeBranchVals(s: var Pairs) =
+  sort(s)
```

Comment:
Would ``nimsets.overlap`` work better here than sorting this list and whatnot?

Reply:
It would be unsafe. 1..2 and 2..3 overlap, but initializing object branch 1..2 with 3 is not good.

The implementation could be simplified using an `IntSet`. 
Some things to consider:
- AFAICT `IntSet` won't work for very large ranges.
- Coverage checking does not work for large range anyway (it overflows), requiring an unnecessary `else` branch. So supporting it is maybe not important. But using a collection of ranges for coverage checking would fix this and produce better error messages for non-enums (see the IntObj test). 
- Either an `IntSet` or ranges could be adapted to support holed-enum.

IMO a collection of ranges is the way to go.